src/RoomInfoListItem.cc

```diff
@@ -77,12 +77,12 @@ RoomInfoListItem::RoomInfoListItem(RoomInfo info, QWidget *parent)
 	avatarLayout_->addWidget(roomAvatar_);
 
 	roomName_ = new QLabel(info_.name(), textWidget_);
-	roomName_->setMaximumSize(parent->width() - 10, 20);
+	roomName_->setMaximumSize(parent->width() - max_height_, 20);
```

Comment:
I reverted the `roomName`, `roomTopic` and `setElidedText` sizing changes, which caused everything to break again:

<img width="1178" alt="screen shot 2017-04-30 at 12 42 46" src="https://cloud.githubusercontent.com/assets/1294269/25564096/e9b2b108-2da2-11e7-9172-89989c607c51.png">

I then re-added the setElidedText change, just to check if it was just the eliding operation which was mis-sized, which improved things a bit but still broke with rooms with a long room name (Megolm test, in this example):

<img width="1178" alt="screen shot 2017-04-30 at 12 44 37" src="https://cloud.githubusercontent.com/assets/1294269/25564100/0c9df0a6-2da3-11e7-8ebf-e49be5cc512c.png">

...so I'm pretty sure that the patch here is correct.  My rationale is that `parent->width()` is returning the width of the whole `RoomInfoListItem`, so you should be sizing the width of the `roomName` and `roomTopic` boxes to that width, minus the width of the avatar (which is set to `max_height_`).

Then again, I know nothing about Qt :D